code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm

```diff
@@ -359,4 +359,387 @@ Difficulty: Very Hard
 /obj/machinery/smartfridge/black_box/default_deconstruction_crowbar()
 	return
 
+///Anomolous Crystal///
+
+/obj/machinery/anomalous_crystal
+	name = "anomalous crystal"
+	desc = "A strange chunk of crystal, being in the presence of it fills you with equal parts excitement and dread."
+	icon = 'icons/obj/lavaland/artefacts.dmi'
+	icon_state = "anomaly_crystal"
+	luminosity = 8
+	burn_state = LAVA_PROOF
+	use_power = 0
+	density = 1
+	languages_spoken = ALL
+	languages_understood = ALL
+	flags = HEAR
+	var/activation_method = "touch"
+	var/activation_damage_type = null
+	var/last_use_timer = 0
+	var/cooldown_add = 30
+	var/list/affected_targets = list()
+	var/activation_sound = 'sound/effects/break_stone.ogg'
+
+/obj/machinery/anomalous_crystal/New()
+	activation_method = pick("touch","laser","bullet","energy","magic","mob_bump","heat","weapon","speech")
+	..()
+
+/obj/machinery/anomalous_crystal/Hear(message, atom/movable/speaker, message_langs, raw_message, radio_freq, spans)
+	..()
+	if(isliving(speaker))
+		ActivationReaction(speaker,"speech")
+
+/obj/machinery/anomalous_crystal/attack_hand(mob/user)
+	..()
+	ActivationReaction(user,"touch")
+
+/obj/machinery/anomalous_crystal/attackby(obj/item/I, mob/user, params)
+	if(I.is_hot())
+		ActivationReaction(user,"heat")
+	else
+		ActivationReaction(user,"weapon")
+	..()
+
+/obj/machinery/anomalous_crystal/bullet_act(obj/item/projectile/P, def_zone)
+	..()
+	if(istype(P, /obj/item/projectile/magic))
+		ActivationReaction(P.firer, "magic", P.damage_type)
+		return
+	ActivationReaction(P.firer, P.flag, P.damage_type)
+
+/obj/machinery/anomalous_crystal/proc/ActivationReaction(mob/user, method, damtype)
+	if(world.time < last_use_timer)
+		return 0
+	if(activation_damage_type && activation_damage_type != damtype)
+		return 0
+	if(method != activation_method)
+		return 0
+	last_use_timer = (world.time + cooldown_add)
+	playsound(user, activation_sound, 100, 1)
+	return 1
+
+/obj/machinery/anomalous_crystal/Bumped(atom/AM as mob|obj)
+	..()
+	if(ismob(AM))
+		ActivationReaction(AM,"mob_bump")
+
+/obj/machinery/anomalous_crystal/random/New()//Just a random crysal spawner for loot
+	var/random_crystal = pick(/obj/machinery/anomalous_crystal/honk, /obj/machinery/anomalous_crystal/theme_warp, /obj/machinery/anomalous_crystal/emitter,
```

Comment:
You can exclude paths from typesof with `typesof(pathX) - pathY` or `typesof(pathX) - listY` and with `subtypesof()`